docs/app/components/usage-link/usage-link.component.ts

```diff
@@ -1,30 +1,52 @@
 import { Usage } from './../../interfaces/Usage';
-import { Input, Renderer2, Component } from '@angular/core';
+import { Input, Renderer2, Component, ElementRef, ViewChild } from '@angular/core';
+import { PopoverDirective, PopoverContainerComponent } from 'ngx-bootstrap/popover';
 
 @Component({
     selector: 'uxd-usage-link',
     templateUrl: './usage-link.component.html',
-    styleUrls: ['./usage-link.component.less']
+    styleUrls: ['./usage-link.component.less'],
+    host: {
+        '(document:click)': 'onClick($event, pop)',
+        '(document:keyup.escape)': 'closePopover()'
+    }
 })
 export class UsageLinkComponent {
 
     @Input() usage: Usage;
-    @Input() pop: string;
 
-    private popoverElement: any;
+    // private popover: PopoverDirective;
+    @ViewChild(PopoverDirective) popover: PopoverDirective;
 
-    constructor(private renderer: Renderer2) {}
+    private popoverElement: HTMLElement;
 
-    onShown(popover: any) {
+    constructor(private renderer: Renderer2, private elementRef: ElementRef) {}
 
-        this.popoverElement = popover._popover._componentRef.location.nativeElement;
+    onShown() {
+        this.popoverElement = (<any>this.popover)._popover._componentRef.location.nativeElement;
 
         this.renderer.setStyle(this.popoverElement, 'style', '300px');
         this.renderer.setStyle(this.popoverElement, 'maxWidth', '300px');
         this.renderer.setStyle(this.popoverElement, 'borderRadius', '0');
         this.renderer.setStyle(this.popoverElement, 'zIndex', '1');
     }
 
+    closePopover () {
+        this.popover.hide();
+    }
+
+    onClick(event: MouseEvent) {
+        let target = event.target as HTMLElement;
```

Comment:
Use the following instead as it uses built in browser functionality and should be more performant:

```typescript
if (!this.popoverElement || !this.popoverElement.contains(event.target as HTMLElement) &&
    !this.elementRef.nativeElement.contains(event.target)) {

    this.popover.hide();
}
```